handler/handler.go

```diff
@@ -4,10 +4,33 @@ import (
 	"net/http"
 	_ "strings"
 
-	_ "durn2.0/durn"
+	"durn2.0/auth"
+	db "durn2.0/database"
+	"durn2.0/durn"
+	"durn2.0/models"
+	"durn2.0/util"
 	_ "github.com/google/uuid"
 )
 
+// isAuthenticatedWithPermissions checks that a the requester is authenticated
+// and has the specified permissions. Writes the proper response to user if not
+func isAuthenticatedWithPermissions(res http.ResponseWriter, req *http.Request, perms []string) bool {
+	if !auth.IsAuthenticated(req.Context()) {
+		err := util.AuthenticationError("Not authorized")
+		util.RequestError(req.Context(), res, err)
+		return false
+	}
+
+	for _, perm := range perms {
+		if err := auth.IsAuthorized(req.Context(), perm); err != nil {
+			util.RequestError(req.Context(), res, err)
+			return false
```

Comment:
Not that it really matters, but I think that this is the only place where you use the `;`-syntax when checking for errors. Is there any reason why it is used here and not in other places? Maybe it would be good to update the others or this one for consistency.

Reply:
I'm not completely sure which way I think should be used. I prefer this way since it is more concise, but It is not possible to use anywhere, e.g. when a function has more than one return value you probably want ot use the non-error one later.

Reply:
Decided to use this stile as much as possible, with the exception being when it would result in a way to long line, in which case `;` is not used in order to improve readability